File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer = 0;
    public int solution(int[] numbers, int target) {
        DFS(0, 0, numbers, target);
        return answer;
    }
    
    public void DFS(int depth, int sum, int[] numbers, int target) {
        if(depth == numbers.length) {
            if(sum == target) {
                answer++;
            }
            return;
        }
        
        DFS(depth + 1, sum + numbers[depth], numbers, target);
        DFS(depth + 1, sum - numbers[depth], numbers, target);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = DFS(numbers, target, 0, 0)
    return answer

def DFS(numbers, target, depth, cur):
    if (depth == len(numbers)):
        if cur == target:
            return 1
        else:
            return 0
    else:
        plus_case = DFS(numbers, target, depth + 1, cur + numbers[depth])
        minus_case = DFS(numbers, target, depth + 1, cur - numbers[depth])
        return plus_case + minus_case
        